MetaMask/metamask-extension
# Migrate remaining design system constants to enums

## Explanation
Currently there are some remaining design system constants that need to be migrated to enums. This PR creates enums for the remaining constants. It also deprecates the constants in favour of the enums and points to a good first issue to encourage contributors to replace them.

*  Enums for design system constants `DISPLAY`, `FLEX_DIRECTION`, `FLEX_WRAP`, `BLOCK_SIZES`, `SEVERITIES` have been created
* `Box` component, documentation and tests updated to favour enums but support constants through unit testing
* Deprecation JSDoc notices added so strike through appears in VS Code as well as good first issue to replace deprecated constants https://github.com/MetaMask/metamask-extension/issues/18714 (Will add constants in the PR once it's merged)
* Fixes: #19130

## Screenshots/Screencaps

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/b4bebdcf-945f-4597-87d7-b28686bf803a

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/03799cc4-a6ba-4782-85fd-c5792b90e59d

## Manual Testing Steps

- Pull this branch
- Do a search of the code base for any of the deprecated constants `DISPLAY`, `FLEX_DIRECTION`, `FLEX_WRAP`, `BLOCK_SIZES`, `SEVERITIES`
- Ensure you have the correct VS Code settings enabled for deprecation indicators
- Check that each constant is deprecated according to JSDoc format

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
